src/ansys/aedt/core/modeler/circuits/primitives_maxwell_circuit.py

```diff
@@ -284,3 +284,87 @@ def create_winding(self, name=None, location=None, angle=0, use_instance_id_netl
         )
         id.set_property("Name", name)
         return id
+
+    @pyaedt_function_handler(compname="name")
+    def create_isin(self, name=None, value=1, location=None, angle=0, use_instance_id_netlist=False):
+        """Create a sinusoidal current source.
+
+        Parameters
+        ----------
+        name : str, optional
+            Name of the current source. The default is ``None``.
+        value : float, optional
+            Value for the amplitude of current. The default is ``1``.
+        location : list of float, optional
+            Position on the X axis and Y axis.
+        angle : float, optional
+            Angle of rotation in degrees. The default is ``0``.
+        use_instance_id_netlist : bool, optional
+            Whether to use the instance ID in the net list or not. The default is ``False``.
+
+        Returns
+        -------
+        :class:`ansys.aedt.core.modeler.cad.object_3dcircuit.CircuitComponent`
+            Circuit Component Object.
+
+        References
+        ----------
+
+        >>> oEditor.CreateComponent
+        """
+        if location is None:
+            location = []
+        id = self.create_component(
```

Comment:
@DaveTwyman create_component returns a component not an id so better to call it in another way. You stated it also in the "Returns" section.

Reply:
@gmalinve , This is the usage of id in the rest of the Maxwell circuit components, for the create_component and in the returns section. @SMoraisAnsys made the same comment earlier on in the history of this PR. I went with consistency throughout the library beating having two library components written in a different way to the other five. 

I'll of course go with your judgement/experience here, but let me know which of these or another you think best
1) Change id usage in all seven components now
2) Implement it differently in these two new components only
3) Implement the id usage in the two new components to be consistent with the other five in the library (As per this PR) and raise another PR for the future to adjust this (And add the remaining missing components whilst we're at it)

Personally I would prefer 3) as it lets me get on with the circuit for the Transformer Toolkit.

Reply:
@DaveTwyman good point. I'd go for the first one, it's just a matter of renaming a variable in all methods but if you don't feel comfortable you can do it in your PR for your methods and then rename the other variables in another PR. 
For me it's just a matter of changing from "id" to "comp" in all methods (or only in yours and remember to do it afterwards for all the others).

Reply:
@gmalinve , Agreed the renaming of 'id' to 'comp' is trivial using ctrl-r in pyaedt. I'll go and do this then. I think it doesn't matter what the name is returned as outside the function but I'll need to go and check this now

Reply:
Renamed 'id' as 'comp_name' throughout 'primitives_maxwell_circuit' file, so marking this as resolved.